### ISSUE TITLE
Assemble load exception for NLog 5.0.0-beta07

### ISSUE THREAD

**Issue by xiaotupansy** [NLog/NLog#2143]
Hi ,

I'm using NLog in an asp.net core app and I want to send logs to elasticsearch.
Everything works fine under NLog 5.0.0-beta06. 
But for NLog 5.0.0-beta07, I found something wrong.
Maybe this is caused by using the same PublicKeyToken to load assembles ?
Following is the nlog internal logs, hope this helps:

2017-06-02 15:33:08.4409 Info Loading assembly: NLog.Web.AspNetCore
2017-06-02 15:33:08.5429 Info Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Extensions.Logging.dll
2017-06-02 15:33:08.5429 Info Loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Extensions.Logging.dll
2017-06-02 15:33:08.5429 Warn Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Extensions.Logging.dll failed! Skipping this file. Exception: System.IO.FileLoadException: Could not load file or assembly 'NLog.Extensions.Logging, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at NLog.Config.ConfigurationItemFactory.BuildDefaultFactory()
2017-06-02 15:33:08.5429 Info Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Targets.ElasticSearch.dll
2017-06-02 15:33:08.5429 Info Loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Targets.ElasticSearch.dll
2017-06-02 15:33:08.5429 Warn Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Targets.ElasticSearch.dll failed! Skipping this file. Exception: System.IO.FileLoadException: Could not load file or assembly 'NLog.Targets.ElasticSearch, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at NLog.Config.ConfigurationItemFactory.BuildDefaultFactory()
2017-06-02 15:33:08.5429 Info Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Web.AspNetCore.dll
2017-06-02 15:33:08.5429 Info Loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Web.AspNetCore.dll
2017-06-02 15:33:08.5429 Warn Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLog.Web.AspNetCore.dll failed! Skipping this file. Exception: System.IO.FileLoadException: Could not load file or assembly 'NLog.Web.AspNetCore, Version=4.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at NLog.Config.ConfigurationItemFactory.BuildDefaultFactory()
2017-06-02 15:33:08.5569 Info Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLogExample.dll
2017-06-02 15:33:08.5569 Info Loading assembly file: C:\inetpub\wwwroot\CoreApp\NLogExample.dll
2017-06-02 15:33:08.5569 Warn Auto loading assembly file: C:\inetpub\wwwroot\CoreApp\NLogExample.dll failed! Skipping this file. Exception: System.IO.FileLoadException: Could not load file or assembly 'NLogExample, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at NLog.Config.ConfigurationItemFactory.BuildDefaultFactory()
2017-06-02 15:33:08.5569 Info Loading assembly: NLog.Targets.ElasticSearch
2017-06-02 15:33:08.7329 Info Adding target BufferingWrapper Target[ElasticSearch](ElasticSearch Target[ElasticSearch])
2017-06-02 15:33:08.7499 Info Adding target File Target[SFile]
2017-06-02 15:33:08.7719 Info Configured from an XML element in C:\inetpub\wwwroot\CoreApp\NLog.config...
2017-06-02 15:33:08.8389 Info Found 47 configuration items
2017-06-02 15:33:08.9940 Info Configuration initialized.
2017-06-02 15:33:08.9940 Info NLog, Version=5.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c
2017-06-02 15:33:09.0170 Info Found 47 configuration items
2017-06-02 15:33:09.0170 Info Loading assembly: NLog.Web.AspNetCore
2017-06-02 15:33:09.0170 Info Loading assembly: NLog.Targets.ElasticSearch
2017-06-02 15:33:09.0170 Info Adding target BufferingWrapper Target[ElasticSearch](ElasticSearch Target[ElasticSearch])
2017-06-02 15:33:09.0250 Info Adding target File Target[SFile]
2017-06-02 15:33:09.0250 Info Configured from an XML element in C:\inetpub\wwwroot\CoreApp\nlog.config...
2017-06-02 15:33:09.0250 Info Closing old configuration.
2017-06-02 15:33:09.1320 Info Found 47 configuration items
2017-06-02 15:33:09.1350 Info Found 47 configuration items

**Comment by xiaotupansy** [comment]
Logs for NLog 5.0.0-beta06:


2017-06-02 15:37:28.4723 Info Loading assembly: NLog.Web.AspNetCore
2017-06-02 15:37:28.5683 Info Loading assembly: NLog.Targets.ElasticSearch
2017-06-02 15:37:28.7213 Info Adding target BufferingWrapper Target[ElasticSearch](ElasticSearch Target[ElasticSearch])
2017-06-02 15:37:28.7384 Info Adding target File Target[SFile]
2017-06-02 15:37:28.7514 Info Configured from an XML element in C:\inetpub\wwwroot\CoreApp\NLog.config...
2017-06-02 15:37:28.8234 Info Found 47 configuration items
2017-06-02 15:37:28.9704 Info Configuration initialized.
2017-06-02 15:37:28.9704 Info NLog, Version=5.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c
2017-06-02 15:37:28.9944 Info Found 47 configuration items
2017-06-02 15:37:28.9944 Info Loading assembly: NLog.Web.AspNetCore
2017-06-02 15:37:28.9944 Info Loading assembly: NLog.Targets.ElasticSearch
2017-06-02 15:37:29.0004 Info Adding target BufferingWrapper Target[ElasticSearch](ElasticSearch Target[ElasticSearch])
2017-06-02 15:37:29.0004 Info Adding target File Target[SFile]
2017-06-02 15:37:29.0004 Info Configured from an XML element in C:\inetpub\wwwroot\CoreApp\nlog.config...
2017-06-02 15:37:29.0004 Info Closing old configuration.
2017-06-02 15:37:29.0604 Info Found 47 configuration items
2017-06-02 15:37:29.0634 Info Found 47 configuration items

**Comment by 304NotModified** [comment]
I think it's a duplicate of https://github.com/NLog/NLog/issues/2120 ?

TL;DR: false positive, you could ignore the error.

**Comment by 304NotModified** [comment]
fixed by #2174 , will be NLog 5 beta 8
